cypress/Pages/BounceRulesDetailed/index.js

```diff
@@ -0,0 +1,153 @@
+import Page from "../page";
+import { Selectors } from "../../../src/components/BounceRuleDetailed/selectors";
+
+class BounceRuleDetailed extends Page {
+  get details() {
+    return cy.get(Selectors.details);
+  }
+
+  get detailsEditable() {
+    return cy.get(Selectors.detailsEditable);
+  }
+
+  get changelog() {
+    return cy.get(Selectors.changelog);
+  }
+
+  get editButton() {
+    return cy.get(Selectors.editButton);
+  }
+
+  get saveButton() {
+    return cy.get(Selectors.saveButton);
+  }
+
+  get cancelButton() {
+    return cy.get(Selectors.cancelButton);
+  }
+
+  get description() {
+    return cy.get(Selectors.description);
+  }
+
+  get bounceId() {
+    return cy.get(Selectors.bounceId);
+  }
+
+  get responseCode() {
+    return cy.get(Selectors.responseCode);
+  }
+
+  get enhancedCode() {
+    return cy.get(Selectors.enhancedCode);
+  }
+
+  get regex() {
+    return cy.get(Selectors.regex);
+  }
+
+  get priority() {
+    return cy.get(Selectors.priority);
+  }
+
+  get bounceAction() {
+    return cy.get(Selectors.bounceAction);
+  }
+
+  get changelogModal() {
+    return cy.get(Selectors.changelogModal);
+  }
+
+  get saveConfirmationModal() {
+    return cy.get(Selectors.saveConfirmationModal);
+  }
+
+  get cancelConfirmationModal() {
+    return cy.get(Selectors.cancelConfirmationModal);
+  }
+
+  get commitInput() {
+    return cy.get(Selectors.commitInput);
+  }
+
+  get confirmSubmit() {
+    return cy.get(Selectors.confirmSubmit);
+  }
+
+  get testChangelog() {
+    return cy.get(`${Selectors.changelog} tbody`).eq(0);
+  }
+
+  get firstChangelog() {
+    return cy.get('[index="0"]');
+  }
+
+  open(ruleId) {
+    super.open(`/bounce_rules/${ruleId}`);
+  }
+
+  createTestRuleAPI(testRule) {
+    return cy.task("createRule", {
+      env: Cypress.env("testEnv"),
+      data: testRule,
+    });
+  }
+
+  deleteBounceRuleAPI(id) {
+    return cy.task("getRules", { env: Cypress.env("testEnv") }).then(res => {
+      if (res) {
+        const isMatchingBounceRule = res.find(
```

Comment:
let's rename this to matchingBounceRule because res.find returns the matching object or undefined/null and later on you're going to do matchingBounceRule.id. is<SomeState> usually denotes only a boolean type and it gets confusing if we expect it only to be a boolean and then access some object property from it. thanks for refactoring this. looking much nicer 😄